src/cls/rgw_gc/cls_rgw_gc.cc

```diff
@@ -276,60 +310,50 @@ static int cls_rgw_gc_queue_remove_entries(cls_method_context_t hctx, bufferlist
     unsigned int index = 0;
     // If data is not empty
     if (op_ret.entries.size()) {
-      for (auto it : op_ret.entries) {
+      for (const auto& it : op_ret.entries) {
         cls_rgw_gc_obj_info info;
         try {
           decode(info, it.data);
         } catch (ceph::buffer::error& err) {
           CLS_LOG(5, "ERROR: cls_rgw_gc_queue_remove_entries(): failed to decode gc info\n");
           return -EINVAL;
         }
+        if (!last_removed_time || *last_removed_time < info.time) {
```

Comment:
while issuing a list call for remove entries, we fetch one extra element to calculate the marker up till which the elements will be deleted. this check will have the last_removed_time set to the expiration time of the last entry, which we dont plan to delete - may result in getting urgent data entry/entries removed which was not intended. I think its better to get the expiration time of the last second entry of the complete list.